testing-framework/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+        accounts_code_cid: Cid,
+    ) -> Account {
+        let rng = &mut rand_chacha::ChaCha8Rng::seed_from_u64(8);
+
+        let priv_key = SecretKey::random(rng);
+
+        let pub_key = PublicKey::from_secret_key(&priv_key);
+        let pub_key_addr =
+            Address::new_secp256k1(&pub_key.serialize()).expect("PublicKey length should be valid");
+
+        let assigned_addr = state_tree
+            .register_new_address(&pub_key_addr)
+            .expect("Should be able to register an account public key on the StateTree");
+
+        let state = fvm::account_actor::State {
+            address: pub_key_addr,
+        };
+
+        let cid = state_tree
+            .store()
+            .put_cbor(&state, Code::Blake2b256)
+            .expect("Should be able to put the Actor State as CBOR");
+
+        let actor_state = ActorState {
+            code: accounts_code_cid,
+            state: cid,
+            sequence: 0,
+            balance: TokenAmount::from_atto(10000),
+            delegated_address: None,
+        };
+
+        state_tree.set_actor(assigned_addr, actor_state);
+        (assigned_addr, pub_key_addr)
+    }
+
+    /// Deploy a new Actor at a given address, provided with a given token balance
+    /// and returns the CodeCID of the installed actor
+    fn deploy_actor_from_bin(
+        &mut self,
+        actor: &WasmActor,
+        balance: TokenAmount,
+    ) -> Result<Address, Error> {
+        let actor_id = rand::random();
+        let actor_address = Address::new_id(actor_id);
+
+        // Put the WASM code into the blockstore.
+        log::debug!("Deploying Actor {} code ", actor.name);
+        let code_cid = self
+            .state_tree
+            .store()
+            .put(
+                Code::Blake2b256,
+                &Block {
+                    codec: IPLD_RAW,
+                    data: &actor.code,
+                },
+            )
+            .setting_err(&actor.name)?;
+
+        // Add code cid to list of deployed contract
+        self.code_cids.push(code_cid);
+
+        // Set the Actor State on the `BlockStore`.
+        Self::set_actor(
+            &mut self.state_tree,
+            &actor.name,
+            &[(); 0],
+            code_cid,
+            actor_id,
+            0,
+            balance,
+        )?;
+
+        Ok(actor_address)
+    }
+
+    /// Deploy the main Actor file into the `StateTree`.
+    pub fn deploy_main_actor(&mut self, actor: WasmActor) -> Result<(), Error> {
+        let address = self.deploy_actor_from_bin(&actor, TokenAmount::zero())?;
+        self.main_actor_id = match address.id() {
+            Ok(id) => Some(id.to_ne_bytes().to_vec()),
+            Err(_) => panic!("Actor Id should be valid"),
+        };
+
+        Ok(())
+    }
+
+    /// Test an Actor on a `MemoryBlockstore`.
+    pub fn test(&mut self, actor: WasmActor, test: WasmActor) -> Result<(), Error> {
+        // TODO: Should we clone the `StateTree` before each test run,
```

Comment:
Having both an `actor` and a `test` parameters here seems redundant, at the very least for the use case we make of them right now (only logging). I think we should:
1. Store names in our tester. 
2. Use the actor ID of the `actor` to refer to it and get its data accordingly. 

_Note: The unique IDs should most likely be actors' ids_

Reply:
yeah agreed, this was from when we thought that actor and test relationship would be 1:1, updated to match our new 1:N usecase ptal Thomas.